### ISSUE TITLE
Intellisense hangs on "Loading..." after working correctly only once

### ISSUE THREAD

**Issue by myounger** [PowerShell/vscode-powershell#2779]
Issue Type: <b>Performance Issue</b>

I've tried everything to get this to work correctly from running with no extensions to completely uninstalling VS Code and reinstalled following steps I found to clear all the extensions and all of the settings. 

The actual issue:
When I begin typing a cmdlet or function name from a custom module, the first occurrence is fine and will auto-complete the parameters. Sometimes the second occurence will work  as well but by the third time, every time, I type out a function from that module the intellisense stops responding and is stuck on "Loading..." for 30-60 seconds. It will eventually clear but this is killing my productivity. It will typically work right away if I kill the Terminal session as well but starting new terminal sessions every few minutes isn't ideal either.

Extension version: 2020.6.1
VS Code version: Code 1.46.1 (cd9ea6488829f560dc949a8b2fb789f3cdc05f5d, 2020-06-17T21:13:20.174Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.88GB (4.36GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	    85	 12944	code main
    0	   180	  2060	   window (Extension: PowerShell Preview - Visual Studio Code)
    0	     6	  4268	     winpty-process
    0	    12	 20940	       console-window-host (Windows internal process)
    0	   181	 22964	       C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command "Import-Module 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.6.1' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\modules' -EnableConsoleRepl -StartupBanner \"=====> PowerShell Preview Integrated Console v2020.6.1 <=====
\" -LogLevel 'Normal' -LogPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\logs\1592840834-ea51b053-7717-47d2-8268-c0a8a60b85821592840765419\EditorServices.log' -SessionDetailsPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\sessions\PSES-VSCode-12944-163499' -FeatureFlags @() "
    0	     6	 13760	     winpty-process
    0	    12	 10044	       console-window-host (Windows internal process)
    0	   202	 21656	       C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	    95	 25452	     extensionHost
    0	    28	 11144	       "C:\Users\myounger\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\myounger\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\jsonServerMain" --node-ipc --clientProcessId=25452
    0	   386	  3856	   gpu-process
    0	    65	  7556	   window (undefined)
    0	    27	 10052	   utility
    0	    60	 12716	   window (undefined)
    0	   160	 14920	   window (● Get-EWaCmdb • Untitled-1 - Visual Studio Code)
    0	     6	 24268	     winpty-process
    0	    12	 13588	       console-window-host (Windows internal process)
    0	   183	 25020	       C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command "Import-Module 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.6.1' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\modules' -EnableConsoleRepl -StartupBanner \"=====> PowerShell Preview Integrated Console v2020.6.1 <=====
\" -LogLevel 'Normal' -LogPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\logs\1592840899-42f3a8ef-0c91-4fec-8f9a-0715882f9e351592840887455\EditorServices.log' -SessionDetailsPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\sessions\PSES-VSCode-12944-956248' -FeatureFlags @() "
    0	    90	 25152	     extensionHost
    0	   192	 22356	   window (Extension: PowerShell Preview - 1592840899-42f3a8ef-0c91-4fec-8f9a-0715882f9e351592840887455 - Visual Studio Code)
    0	    14	  6044	     watcherService 
    0	    11	 10092	       console-window-host (Windows internal process)
    0	    36	 11236	     searchService
    0	     6	 13824	     winpty-process
    0	    12	  8792	       console-window-host (Windows internal process)
    0	   203	 23036	       C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command "Import-Module 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.6.1' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\modules' -EnableConsoleRepl -StartupBanner \"=====> PowerShell Preview Integrated Console v2020.6.1 <=====
\" -LogLevel 'Normal' -LogPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\logs\1592841313-b39500ac-7032-4258-801c-fbf10160e6fb1592841015051\EditorServices.log' -SessionDetailsPath 'c:\Users\myounger\.vscode\extensions\ms-vscode.powershell-preview-2020.6.1\sessions\PSES-VSCode-12944-808360' -FeatureFlags @() "
    0	    92	 23448	     extensionHost
    0	    28	 22124	       "C:\Users\myounger\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\myounger\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\jsonServerMain" --node-ipc --clientProcessId=23448
    0	    77	 22424	   shared-process
    0	    13	 22984	   electron-crash-reporter
    0	   164	 25188	   window (EditorServices.log - 1592841313-b39500ac-7032-4258-801c-fbf10160e6fb1592841015051 - Visual Studio Code)
    0	    34	  6396	     searchService
    0	    83	 18852	     extensionHost
    0	    14	 21532	     watcherService 
    0	    11	 15200	       console-window-host (Windows internal process)
    0	    73	 25476	   window (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (Extension: PowerShell Preview - 1592840899-42f3a8ef-0c91-4fec-8f9a-0715882f9e351592840887455 - Visual Studio Code)
|  Window (EditorServices.log - 1592841313-b39500ac-7032-4258-801c-fbf10160e6fb1592841015051 - Visual Studio Code)
|    Folder (1592841313-b39500ac-7032-4258-801c-fbf10160e6fb1592841015051): 3 files
|      File types: log(3)
|      Conf files:
|    Folder (1592840899-42f3a8ef-0c91-4fec-8f9a-0715882f9e351592840887455): 3 files
|      File types: log(3)
|      Conf files:;
```

</details>

**Comment by myounger** [comment]
Some more information. This is broken on my primary development machine. However, I do have another one that is not having this issue. Here is what I found.

I turned on diagnostic mode for the editor services log. There is a distinct difference between the broken machine and the one that works.

Broken:
```
[Info  - 12:38:41 PM] OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finished: Routing Request (140) textDocument/completion in 21041ms | Id='140' Method='textDocument/completion' ElapsedMilliseconds='21041'
[Info  - 12:38:41 PM] OmniSharp.Extensions.JsonRpc.InputHandler: Request 140 was cancelled | Id='140'
```

Works:
```
[Info  - 12:40:11 PM] OmniSharp.Extensions.LanguageServer.Shared.LspRequestRouter: Finished: Routing Request (87) textDocument/documentSymbol in 0ms | Id='87' Method='textDocument/documentSymbol' ElapsedMilliseconds='0'
[Info  - 12:40:11 PM] OmniSharp.Extensions.JsonRpc.InputHandler: Finished: Processing request textDocument/documentSymbol 87 in 0ms | Method='textDocument/documentSymbol' ResponseId='87' ElapsedMilliseconds='0'
```

You can see on the machine that breaks, it is running for 21 seconds before it "completes" and then says the request was cancelled. For the machine that works, this is a 0ms call and it returns fine.

**Comment by SydneyhSmith** [comment]
Thanks for reporting this and providing detailed information @myounger this looks like a duplicate of #2745

**Comment by myounger** [comment]
Yeah, I agree this looks like a duplicate. Feel free to close this is if you don't need anything additional from me.
